### PR TITLE
Paddle OCR with Ernie

### PR DESCRIPTION
**Pull request by forBlank**
modeling PPOCRVL & SiglipViT, create OCR workflow & only DP dataflow

### CODE REVIEW

File: data_processor/image_preprocessor/image_preprocessor_siglip.py [hunk 1]
```diff
@@ -0,0 +1,542 @@
...
+        augmentations.append(RandomApply([color_jitter], p=color_jitter_p))
+
+    if jpeg_p > 0:
+        jpeg_transform = JpegCompression(quality_range=jpeg_quality_range)
+        augmentations.append(RandomApply([jpeg_transform], p=jpeg_p))
+
+    return transforms.Compose(augmentations)
+
+
+image_augmentation = get_ocr_augmentations(
```
Comment (nepeplwu) [inline data_processor/image_preprocessor/image_preprocessor_siglip.py:197]: 不需要开放给用户进行设置吗

File: ernie/dataset/dist_data_loader.py [hunk 2]
```diff
@@ -207,6 +210,7 @@ def __init__(
             lambda x: x,  # collate_fn,
             num_workers,
             use_buffer_reader,
+            reader_buffer_size,
```
Comment (BossPi) [inline ernie/dataset/dist_data_loader.py]: Dataloader能接受这个参数吗？

File: ernie/configuration_ocr.py [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+from paddleformers.transformers.configuration_utils import PretrainedConfig
+from .siglip.modeling import PPOCRVisionConfig
+
+
+class PPOCRVLConfig(PretrainedConfig):
+    model_type = "ppocrvl"
+    keys_to_ignore_at_inference = ["past_key_values"]
+    sub_configs = {"vision_config": PPOCRVisionConfig}
+
+    base_model_tp_plan = {
```
Comment (nepeplwu) [inline ernie/configuration_ocr.py]: 这个是否有必要保留？

File: ernie/dataset/vl_sft_reader/data_utils.py [hunk 4]
```diff
@@ -241,13 +244,13 @@ def merge_fn_group_batch(
...
             to_concat = [b[k] for b in batch if b[k] is not None]
             ret[k] = np.concatenate(to_concat, axis=0)
-            if pad_to_max_seqlen:
-                tmp = max(0, pad_to_max_seqlen * len(batch) - ret[k].shape[0])
-                if tmp > 0:
-                    ret[k] = np.concatenate(
-                        [ret[k], np.zeros([tmp, 3])], axis=0
-                    ).astype("int64")
-        elif k in ["data_id", "part_id", "src_id"]:
+            # if pad_to_max_seqlen:
```
Comment (nepeplwu) [inline ernie/dataset/vl_sft_reader/data_utils.py]: 如果不需要的代码建议直接删掉

**Comment by nepeplwu** [comment]
`examples/configs/ERNIE-4.5-0.3B/sft/run_ocr_sft_8k.yaml` -> `examples/configs/PaddleOCR-VL/sft/run_ocr_sft_8k.yaml`

File: ernie/dataset/dist_data_loader.py (hunk 2 above)
Comment (forBlank) [reply ernie/dataset/dist_data_loader.py]: paddle develop 版本中 Dataloader 可以接受这个参数 https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/io/reader.py#L470

File: ernie/dataset/vl_sft_reader/data_utils.py (hunk 4 above)
Comment (forBlank) [reply ernie/dataset/vl_sft_reader/data_utils.py]: 此处个人认为不需要对 image的 grid_thw 进行 padding，后续将删除

File: ernie/configuration_ocr.py (hunk 3 above)
Comment (forBlank) [reply ernie/configuration_ocr.py]: 可以讨论一下 PaddleFormers 是否需要保留这些 attribute

File: data_processor/image_preprocessor/image_preprocessor_siglip.py (hunk 1 above)
Comment (forBlank) [reply data_processor/image_preprocessor/image_preprocessor_siglip.py:197]: 目前使用的 augmentation 是 Paddle OCR 固定的，没有开放用户设置接口

File: ernie/dataset/dist_data_loader.py (hunk 2 above)
Comment (nepeplwu) [reply ernie/dataset/dist_data_loader.py]: 需要确定下最小支持的paddle版本，另外这个功能的必要性大吗？

File: erniekit/train/ocr_vl_sft/pretraining_trainer.py [hunk 5]
```diff
@@ -0,0 +1,1675 @@
...
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""The trainer class used in training."""
+
+__all__ = [
+    "PretrainingTrainer",
```
Comment (nepeplwu) [inline erniekit/train/ocr_vl_sft/pretraining_trainer.py:20]: 现在支持预训练吗？

File: erniekit/hparams/preprocess_args.py [hunk 6]
```diff
@@ -141,6 +141,16 @@ class ImageModificationProcessorArguments(BasePreprocessArguments):
     sft_shift_by_one: bool = field(
         default=False, metadata={"help": "SFT data_processor shift-by-one"}
     )
+    sft_replace_ids: bool = field(
+        default=False,
+        metadata={"help": "SFT replace sep_token with eos_token in input_ids"},
```
Comment (nepeplwu) [inline erniekit/hparams/preprocess_args.py:146]: 为什么OCR需要添加这三个flag？

File: ernie/siglip/modeling_pp.py [hunk 7]
```diff
@@ -0,0 +1,94 @@
...
+import paddle
+from paddleformers.utils.log import logger
+
+from .modeling import SiglipVisionModel, BaseModelOutputWithPooling
+from ..sequence_parallel_utils import (
+    mark_as_sequence_parallel_parameter,
+)
+
+
+class SiglipVisionModelPipe(SiglipVisionModel):
```
Comment (nepeplwu) [inline ernie/siglip/modeling_pp.py]: 这个组网为什么需要实现pp

File: erniekit/train/ocr_vl_sft/pretraining_trainer.py (hunk 5 above)
Comment (forBlank) [reply erniekit/train/ocr_vl_sft/pretraining_trainer.py:20]: 这里 trainer.py 的 SFTTrainer 需要继承 pretraining_trainer.py 中的 PretrainingTrainer，理论上不使用 chat_template 即可支持预训练

File: erniekit/hparams/preprocess_args.py (hunk 6 above)
Comment (forBlank) [reply erniekit/hparams/preprocess_args.py:146]: 为了完全对齐 PaddleOCR 的训练数据流

File: ernie/siglip/modeling_pp.py (hunk 7 above)
Comment (forBlank) [reply ernie/siglip/modeling_pp.py]: 这部分后续会删除

File: examples/configs/PaddleOCR-VL/sft/run_ocr_sft_8k.yaml [hunk 8]
```diff
@@ -0,0 +1,92 @@
...
+max_seq_len: 1536
+num_samples_each_epoch: 6000000
+use_pic_id: False
+sft_replace_ids: True
+sft_image_normalize: True
+sft_image_rescale: True
+image_dtype: "float32"
+
+### model
+model_name_or_path: baidu/ERNIE-4.5-0.3B-Paddle
```
Comment (Sunting78) [inline examples/configs/PaddleOCR-VL/sft/run_ocr_sft_8k.yaml]: 这里是语言模型？不是PaddleOCR-VL-0.9B嘛？

File: requirements/gpu/requirements.txt [hunk 9]
```diff
@@ -11,3 +11,4 @@ triton==3.3
 use-triton-in-paddle
 tensorboardX
 ujson
+einops
```
Comment (nepeplwu) [inline requirements/gpu/requirements.txt:14]: 为什么会引入这种依赖？

File: docs/erniekit.md [hunk 10]
```diff
@@ -7,6 +7,13 @@
 
 ## 📣 Recent updates
 
+**[2025-10] 🔥 Released ERNIEKit v1.4:**
+
+- **New Features**
+    - VL Modle Training: Support SFT for PaddleOCR-VL-0.9B model. More details in [PaddleOCR-VL-0.9B SFT](./paddleocr_vl_sft.md).
```
Comment (nepeplwu) [inline docs/erniekit.md]: PaddleOCR-VL-0.9B引用链接到huggingface

File: docs/paddleocr_vl_sft.md [hunk 11]
```diff
@@ -0,0 +1,48 @@
+# PaddleOCR-VL-0.9B SFT
```
Comment (nepeplwu) [inline docs/paddleocr_vl_sft.md:3]: 增加一个中文版，在这个页面链接进行切换

File: requirements/gpu/requirements.txt (hunk 9 above)
Comment (forBlank) [reply requirements/gpu/requirements.txt:14]: modeling_paddleocr_vl.py 里有用到 einops.rearrange

Comment (nepeplwu) [reply requirements/gpu/requirements.txt:14]: 确认下看看是否有替换实现，尽量减少新包的依赖

File: ernie/configuration_ocr.py (hunk 3 above)
Comment (forBlank) [reply ernie/configuration_ocr.py]: 已删除
